python/cugraph/structure/graph.py

```diff
@@ -565,945 +277,394 @@ def from_numpy_matrix(self, np_matrix):
-
-        Returns
-        -------
-        df : cudf.DataFrame
-            GPU DataFrame of size N (the default) or the size of the given
-            vertices (vertex_subset) containing the degrees. The ordering is
-            relative to the adjacency list, or that given by the specified
-            vertex_subset.
-
-            df['vertex'] : cudf.Series
-                The vertex IDs (will be identical to vertex_subset if
-                specified).
-            df['in_degree'] : cudf.Series
-                The in-degree of the vertex.
-            df['out_degree'] : cudf.Series
-                The out-degree of the vertex.
-
-        Examples
-        --------
-        >>> M = cudf.read_csv('datasets/karate.csv', delimiter=' ',
-        >>>                   dtype=['int32', 'int32', 'float32'], header=None)
-        >>> G = cugraph.Graph()
-        >>> G.from_cudf_edgelist(M, '0', '1')
-        >>> df = G.degrees([0,9,12])
-
-        """
-        if self.distributed:
-            raise Exception("Not supported for distributed graph")
-        (
-            vertex_col,
-            in_degree_col,
-            out_degree_col,
-        ) = graph_primtypes_wrapper._degrees(self)
-
-        df = cudf.DataFrame()
-        df["vertex"] = vertex_col
-        df["in_degree"] = in_degree_col
-        df["out_degree"] = out_degree_col
-
-        if self.renumbered is True:
-            df = self.unrenumber(df, "vertex")
-
-        if vertex_subset is not None:
-            df = df[df['vertex'].isin(vertex_subset)]
-
-        return df
-
-    def _degree(self, vertex_subset, x=0):
-        vertex_col, degree_col = graph_primtypes_wrapper._degree(self, x)
-        df = cudf.DataFrame()
-        df["vertex"] = vertex_col
-        df["degree"] = degree_col
-
-        if self.renumbered is True:
-            df = self.unrenumber(df, "vertex")
-
-        if vertex_subset is not None:
-            df = df[df['vertex'].isin(vertex_subset)]
-
-        return df
+        return self.properties.isolated_vertices
 
     def to_directed(self):
         """
         Return a directed representation of the graph.
         This function sets the type of graph as DiGraph() and returns the
         directed view.
-
         Returns
         -------
         G : DiGraph
             A directed graph with the same nodes, and each edge (u,v,weights)
             replaced by two directed edges (u,v,weights) and (v,u,weights).
-
         Examples
         --------
         >>> M = cudf.read_csv('datasets/karate.csv', delimiter=' ',
         >>>                   dtype=['int32', 'int32', 'float32'], header=None)
         >>> G = cugraph.Graph()
         >>> G.from_cudf_edgelist(M, '0', '1')
         >>> DiG = G.to_directed()
-
         """
-        if self.distributed:
-            raise Exception("Not supported for distributed graph")
-        if type(self) is DiGraph:
+        if self.graph_properties.directed is True:
```

Comment:
maybe use `hasattr` for consistency?

Reply:
https://github.com/rapidsai/cugraph/pull/1404#discussion_r607573439